packages/venia-concept/src/components/MiniCart/editItem.js

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect } from 'react';
+import { bool, func, object } from 'prop-types';
+
+import { useQuery } from '@magento/peregrine';
+
+import { loadingIndicator } from 'src/components/LoadingIndicator';
+import PRODUCT_DETAILS from 'src/queries/getProductDetailByName.graphql';
+
+import CartOptions from './cartOptions';
+
+const EditItem = props => {
+    // Props.
+    const { endEditItem, isUpdatingItem, item, updateItemInCart } = props;
+
+    // State / Hooks.
+    const [queryResult, queryApi] = useQuery(PRODUCT_DETAILS);
+    const { data, error } = queryResult;
+    const { runQuery, setLoading } = queryApi;
+
+    // Members.
+    const itemHasOptions = item && item.options && item.options.length > 0;
+
+    // Run the query once on mount and again whenever the
+    // item being edited changes.
+    useEffect(() => {
+        // We must have an item with options.
+        if (!(item && itemHasOptions)) {
+            return;
+        }
+
+        setLoading(true);
+        runQuery({
+            variables: {
+                name: item.name,
+                onServer: false
+            }
+        });
+    }, [item]);
+
+    /*
+     * Render.
+     */
+
+    // We must have an item to edit.
+    if (!item) {
+        // TODO: log an error? Pop a toast?
+        return null;
+    }
+
+    // This is a non-configurable item, just set configItem to an empty object.
+    if (!itemHasOptions) {
```

Comment:
We have the item in the cart and it may or may not have `options` - things like fashion size "M" and fashion color = "Blue".

We're about to open the edit screen (to change options) so the query is _for the rest of the options_. In other words, the other fashion sizes and fashion colors that the user could possibly choose for this item.

Reply:
So the cart item used to render is a subset of the product? OK. I get that.

Reply:
Right, the `error` and `loading` states only apply if we have a configurable item